wallet/chainntfns.go

```diff
@@ -888,18 +895,26 @@ func (w *Wallet) handleWinningTickets(blockHash *chainhash.Hash,
 	// be fixed somehow, but this should be stable for networks that
 	// are voting at normal block speeds.
 
-	// Only consider tickets owned by this wallet.
-	ticketHashes := w.StakeMgr.OwnTickets(winningTicketHashes)
-	if len(ticketHashes) == 0 {
-		return nil
-	}
-
-	votes := make([]*wire.MsgTx, len(ticketHashes))
+	var ticketHashes []*chainhash.Hash
+	var votes []*wire.MsgTx
 	voteBits := w.VoteBits()
 	err = walletdb.View(w.db, func(dbtx walletdb.ReadTx) error {
+		txmgrNs := dbtx.ReadBucket(wtxmgrNamespaceKey)
+
+		// Only consider tickets owned by this wallet.
+		ticketHashes = selectOwnedTickets(w, dbtx, winningTicketHashes)
+		if len(ticketHashes) == 0 {
+			return nil
+		}
+
+		votes = make([]*wire.MsgTx, len(ticketHashes))
+
 		addrmgrNs := dbtx.ReadBucket(waddrmgrNamespaceKey)
 		for i, ticketHash := range ticketHashes {
-			ticketPurchase, err := w.StakeMgr.TicketPurchase(dbtx, ticketHash)
+			ticketPurchase, err := w.TxStore.Tx(txmgrNs, ticketHash)
+			if err != nil || ticketPurchase == nil {
```

Comment:
Right, the situation where ticketPurchase == nil after asking TxStore, so then you ask StakeMgr